pkg/apis/configuration/validation/validation.go

```diff
@@ -156,6 +159,41 @@ func validateUpstreams(upstreams []v1alpha1.Upstream, fieldPath *field.Path, isP
 	return allErrs, upstreamNames
 }
 
+var validVariableParams = map[string]bool{
+	"error":          true,
+	"timeout":        true,
+	"invalid_header": true,
+	"http_500":       true,
+	"http_502":       true,
+	"http_503":       true,
+	"http_504":       true,
+	"http_403":       true,
+	"http_404":       true,
+	"http_429":       true,
+	"non_idempotent": true,
+	"off":            true,
+}
+
+// ValidateNextUpstream checks the values given for passing queries to a upstream
```

Comment:
You don't need to comment private functions. However, if you do it, I think is better to change the capital `V` for `v` for consistency.